File: tests/leetcode/x/python/0679.py
import sys

EPS = 1e-6


def solve(cards: list[int]) -> bool:
    nums = [float(x) for x in cards]

    def dfs(arr: list[float]) -> bool:
        if len(arr) == 1:
            return abs(arr[0] - 24.0) < EPS
        n = len(arr)
        for i in range(n):
            for j in range(i + 1, n):
                rest = [arr[k] for k in range(n) if k != i and k != j]
                a, b = arr[i], arr[j]
                candidates = [a + b, a * b, a - b, b - a]
                if abs(b) > EPS:
                    candidates.append(a / b)
                if abs(a) > EPS:
                    candidates.append(b / a)
                for value in candidates:
                    if dfs(rest + [value]):
                        return True
        return False

    return dfs(nums)


def main() -> None:
    lines = [line.strip() for line in sys.stdin.read().splitlines() if line.strip()]
    if not lines:
        return
    out = ["true" if solve(list(map(int, line.split()))) else "false" for line in lines[1:1 + int(lines[0])]]
    sys.stdout.write("\n\n".join(out))


if __name__ == "__main__":
    main()

File: tests/leetcode/x/go/0679.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

const eps = 1e-6

func solve(cards []int) bool {
	nums := make([]float64, len(cards))
	for i, x := range cards {
		nums[i] = float64(x)
	}
	var dfs func([]float64) bool
	dfs = func(arr []float64) bool {
		if len(arr) == 1 {
			return math.Abs(arr[0]-24.0) < eps
		}
		n := len(arr)
		for i := 0; i < n; i++ {
			for j := i + 1; j < n; j++ {
				rest := make([]float64, 0, n-1)
				for k := 0; k < n; k++ {
					if k != i && k != j {
						rest = append(rest, arr[k])
					}
				}
				a, b := arr[i], arr[j]
				cands := []float64{a + b, a * b, a - b, b - a}
				if math.Abs(b) > eps {
					cands = append(cands, a/b)
				}
				if math.Abs(a) > eps {
					cands = append(cands, b/a)
				}
				for _, v := range cands {
					next := append(rest, v)
					if dfs(next) {
						return true
					}
				}
			}
		}
		return false
	}
	return dfs(nums)
}

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()
	var t int
	if _, err := fmt.Fscan(in, &t); err != nil {
		return
	}
	for i := 0; i < t; i++ {
		cards := make([]int, 4)
		for j := 0; j < 4; j++ {
			fmt.Fscan(in, &cards[j])
		}
		if i > 0 {
			fmt.Fprintln(out)
			fmt.Fprintln(out)
		}
		if solve(cards) {
			fmt.Fprint(out, "true")
		} else {
			fmt.Fprint(out, "false")
		}
	}
}
